src/app/clusters/ota-requestor/DefaultOTARequestor.cpp

```diff
@@ -670,17 +673,25 @@ void DefaultOTARequestor::RecordNewUpdateState(OTAUpdateStateEnum newState, OTAC
 
     if ((newState == OTAUpdateStateEnum::kIdle) && (mCurrentUpdateState != OTAUpdateStateEnum::kIdle))
     {
-        IdleStateReason idleStateReason = MapErrorToIdleStateReason(error);
-
-        // Inform the driver that the core logic has entered the Idle state
-        mOtaRequestorDriver->HandleIdleStateEnter(idleStateReason);
+        idleStateReason      = MapErrorToIdleStateReason(error);
+        handleIdleStateEnter = true;
```

Comment:
If this starts off as true, why set it true again?

or if this is for clarity, how about a final else branch after below to make it super clear?

naming wise, is this `idleStateEntered`? Why is the declare so high up like C-style? reduced scope is easier to read. same for idleStateReason.

```cpp
const bool isEnteringIdle = (newState == OTAUpdateStateEnum::kIdle)
                            && (mCurrentUpdateState != OTAUpdateStateEnum::kIdle);
```

Reply:
Got rid of this boolean as I was updating the code per your comment as I realize there's 3 states here and not 2.  The 3rd case is when we're neither entering nor exiting idle state.